cmd/ingester/app/consumer/consumer_test.go

```diff
@@ -52,212 +49,283 @@ func TestConstructor(t *testing.T) {
-		Name:  "sarama-consumer.partitions-held",
-		Value: 1,
-	})
+	consumer, err := New(consumerParams, WithWaitReady(true))
+	assert.NoError(t, err)
 
-	mp.AssertExpectations(t)
-	// Ensure that the partition consumer was updated in the map
-	assert.Equal(t, saramaPartitionConsumer.HighWaterMarkOffset(),
-		undertest.partitionIDToState[partition].partitionConsumer.HighWaterMarkOffset())
-	undertest.Close()
+	sw.t = t
 
-	localFactory.AssertCounterMetrics(t, metricstest.ExpectedMetric{
-		Name:  "sarama-consumer.partitions-held",
-		Value: 0,
-	})
+	consumer.Start()
 
-	partitionTag := map[string]string{"partition": fmt.Sprint(partition)}
-	localFactory.AssertCounterMetrics(t, metricstest.ExpectedMetric{
-		Name:  "sarama-consumer.messages",
-		Tags:  partitionTag,
-		Value: 1,
-	})
-	localFactory.AssertGaugeMetrics(t, metricstest.ExpectedMetric{
-		Name:  "sarama-consumer.current-offset",
-		Tags:  partitionTag,
-		Value: int(msgOffset),
-	})
-	localFactory.AssertGaugeMetrics(t, metricstest.ExpectedMetric{
-		Name: "sarama-consumer.offset-lag",
-		Tags: partitionTag,
-		// Prior to sarama v1.31.0 this would be 0, it's unclear why this changed.
-		// v=1 seems to be correct because high watermark in mock is incremented upon
-		// consuming the message, and func HighWaterMarkOffset() returns internal value
-		// (already incremented) + 1, so the difference is always 2, and we then
-		// subtract 1 from it.
-		Value: 1,
-	})
-	localFactory.AssertCounterMetrics(t, metricstest.ExpectedMetric{
-		Name:  "sarama-consumer.partition-start",
-		Tags:  partitionTag,
-		Value: 1,
-	})
+	t.Log("Consumer is ready and wait message")
+
+	<-sw.spanWriteCh
+
+	consumer.Close()
+
+	t.Logf("Consumer all logs: %s", logBuf.String())
 }
 
-func TestSaramaConsumerWrapper_start_Errors(t *testing.T) {
-	localFactory := metricstest.NewFactory(0)
+func newSampleSpan(traceID model.TraceID, spanID model.SpanID) (string, error) {
+	sampleTags := model.KeyValues{
+		model.String("someStringTagKey", "someStringTagValue"),
+	}
+	sampleSpan := &model.Span{
+		TraceID:       traceID,
+		SpanID:        spanID,
+		OperationName: "someOperationName",
+		References: []model.SpanRef{
+			{
+				TraceID: traceID,
+				SpanID:  spanID,
+				RefType: model.ChildOf,
+			},
+		},
+		Flags:     model.Flags(1),
+		StartTime: model.EpochMicrosecondsAsTime(55555),
+		Duration:  model.MicrosecondsAsDuration(50000),
+		Tags:      sampleTags,
+		Logs: []model.Log{
+			{
+				Timestamp: model.EpochMicrosecondsAsTime(12345),
+				Fields:    sampleTags,
+			},
+		},
+		Process: &model.Process{
+			ServiceName: "someServiceName",
+			Tags:        sampleTags,
+		},
+	}
 
-	saramaConsumer := smocks.NewConsumer(t, &sarama.Config{})
-	mc := saramaConsumer.ExpectConsumePartition(topic, partition, msgOffset)
-	mc.ExpectErrorsDrainedOnClose()
+	m := &jsonpb.Marshaler{}
+	msg, err := m.MarshalToString(sampleSpan)
+	if err != nil {
+		return "", err
+	}
```

Comment:
nit: 
```
func newSampleSpan(traceID model.TraceID, spanID model.SpanID) (string, error)
```

can be simplified to:

```
func newSampleSpan(traceID model.TraceID, spanID model.SpanID) string
```

by adding a `require.NoError(err)` here. 

It means its usage could also be simplified from:
```
msg1, err := newSampleSpan(model.NewTraceID(1, 2), model.NewSpanID(1))
require.NoError(t, err)
msg2, err := newSampleSpan(model.NewTraceID(2, 3), model.NewSpanID(2))
require.NoError(t, err)
```

to:

```
msg1 := newSampleSpan(model.NewTraceID(1, 2), model.NewSpanID(1))
msg2 := newSampleSpan(model.NewTraceID(2, 3), model.NewSpanID(2))
```

Reply:
I'll fix it later